taskfile/cache_internal_test.go

```diff
@@ -0,0 +1,99 @@
+package taskfile
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/go-task/task/v3/internal/logger"
+)
+
+func ExampleWithTTL() {
+	c, _ := NewCache(os.TempDir(), WithTTL(2*time.Minute+30*time.Second))
+
+	fmt.Println(c.ttl)
+	// Output: 2m30s
+}
+
+var discardLogger = &logger.Logger{
+	Stdout: io.Discard,
+	Stderr: io.Discard,
+}
+
+func primeNewCache(t *testing.T, tempDir string, cacheOpts ...CacheOption) (*Cache, *FileNode) {
+	t.Helper()
+
+	cache, err := NewCache(tempDir, cacheOpts...)
+	require.NoErrorf(t, err, "creating new cache in temporary directory '%s'", tempDir)
+
+	// Prime the temporary cache directory with a basic Taskfile.
+	filename := "Taskfile.yaml"
+	srcTaskfile := filepath.Join("testdata", filename)
+	dstTaskfile := filepath.Join(cache.dir, filename)
+
+	taskfileBytes, err := os.ReadFile(srcTaskfile)
+	require.NoErrorf(t, err, "reading from testdata Taskfile (%s)", srcTaskfile)
+
+	err = os.WriteFile(dstTaskfile, taskfileBytes, 0o640)
+	require.NoErrorf(t, err, "writing to temporary Taskfile (%s)", dstTaskfile)
+
+	// Create a new file node in the cache, with the entrypoint copied above.
+	fileNode, err := NewFileNode(discardLogger, dstTaskfile, cache.dir)
+	require.NoError(t, err, "creating new file node")
+
+	return cache, fileNode
+}
+
+func TestCache(t *testing.T) {
+	cache, fileNode := primeNewCache(t, t.TempDir())
+
+	// Attempt to read from cache, then write, then read again.
+	_, err := cache.read(fileNode)
+	require.ErrorAs(t, err, &os.ErrNotExist, "reading from cache before writing should match error type")
```

Comment:
I'm surprised, I would have expected an ErrorIs here


```suggestion
	require.ErrorIs(t, err, os.ErrNotExist, "reading from cache before writing should match error type")
```

os.ErrNotExist is a var not a type.

You maybe have used ErrorAs with `os.PathError`

Reply:
I think I was using `ErrorAs` for a different error type, before a refactor, yes. Will switch it over, thanks.

Reply:
b9ad0bff3c213f87b75aed719bf8adce8d0f9540 will resolve.